pkg/nuctl/command/common/patchmanifest.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2023 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"sync"
+
+	"github.com/nuclio/logger"
+)
+
+type PatchManifest struct {
+	*patchManifest
+	lock sync.Mutex
+}
+
+type patchManifest struct {
+	Success []string                   `json:"success,omitempty"`
+	Skipped []string                   `json:"skipped,omitempty"`
+	Failed  map[string]FailDescription `json:"failed,omitempty"`
+}
+
+type FailDescription struct {
+	Err       string `json:"error,omitempty"`
+	Retryable bool   `json:"retryable"`
+}
```

Comment:
No reason for this struct (and its attributes) to be Public.

But in general - put public objects above private objects.

Reply:
@TomerShor if we make them private, we won't be able to export them in json, this is a main reason

Reply:
Oh that's right.
Then just move it above the `patchManifest` private object